Dockerfile.process-sandbox

```
@@ -25,15 +25,39 @@ RUN apt-get update \
+# Rust toolchain, installed as the sandbox user under the workspace-relative
+# HOME so it survives container-rm/recreate only via the bind-mounted
+# workspace (matches the design's "HOME=/workspace/.home so caches/dotfiles
+# persist" decision) — installed at build time into a location Cargo can
+# still find once HOME is redirected to /workspace/.home at runtime via
+# CARGO_HOME/RUSTUP_HOME pointed at a workspace-relative path is NOT done at
+# build time (the workspace doesn't exist yet); instead rustup installs to
+# the image's default /home/sandbox/.cargo, and the container launch env
+# additionally sets CARGO_HOME=/workspace/.home/.cargo, RUSTUP_HOME=/workspace/.home/.rustup
+# with a first-run copy step in the entrypoint script — see entrypoint change below.
+USER sandbox
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile minimal
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)